### PR TITLE
Removing all trees by container ID if tree ID is empty in `pilorama.Forest.TreeDrop`

### LINKED ISSUE #1940: Allow removing all trees by container ID
[Cleaning line](https://github.com/nspcc-dev/neofs-node/blob/7d456cb4d0c5381020b4bd76bfc9c5ab5683f90a/cmd/neofs-node/tree.go#L58) looks buggy. I doubt the `pilorama` [removes](https://github.com/nspcc-dev/neofs-node/blob/7d456cb4d0c5381020b4bd76bfc9c5ab5683f90a/pkg/local_object_storage/pilorama/boltdb.go#L611) all existing trees.

### PR DESCRIPTION
**Pull request by acid-ant**
Close #1940

Signed-off-by: Anton Nikiforov <an.nikiforov@yadro.com>

### CODE REVIEW

File: pkg/local_object_storage/pilorama/boltdb.go [hunk 1]
```diff
@@ -625,8 +625,20 @@ func (t *boltForest) TreeGetOpLog(cid cidSDK.ID, treeID string, height uint64) (
 }
 
 // TreeDrop implements the pilorama.Forest interface.
+// In case of empty treeID drops all trees related to container.
```
Comment (cthulhu-rider) [inline pkg/local_object_storage/pilorama/boltdb.go:628]: Lets maybe move it to `pilorama.Forest` interface?

Comment (acid-ant) [reply pkg/local_object_storage/pilorama/boltdb.go]: Agree. Moved.

File: pkg/local_object_storage/pilorama/forest.go [hunk 2]
```diff
@@ -183,13 +184,21 @@ func (f *memoryForest) TreeGetOpLog(cid cidSDK.ID, treeID string, height uint64)
...
 // TreeDrop implements the pilorama.Forest interface.
 func (f *memoryForest) TreeDrop(cid cidSDK.ID, treeID string) error {
-	fullID := cid.String() + "/" + treeID
-	_, ok := f.treeMap[fullID]
-	if !ok {
-		return ErrTreeNotFound
+	cidStr := cid.String()
+	if treeID == "" {
+		for k := range f.treeMap {
+			if bytes.HasPrefix([]byte(k), []byte(cidStr)) {
```
Comment (cthulhu-rider) [inline pkg/local_object_storage/pilorama/forest.go:190]: We can use same functionality from `strings` package.

**Review by cthulhu-rider — APPROVED** [review]

File: pkg/local_object_storage/pilorama/forest.go (hunk 2 above)
Comment (acid-ant) [reply pkg/local_object_storage/pilorama/forest.go]: Thanks, fixed.

File: pkg/local_object_storage/pilorama/boltdb.go [hunk 3]
```diff
@@ -627,6 +627,17 @@ func (t *boltForest) TreeGetOpLog(cid cidSDK.ID, treeID string, height uint64) (
 // TreeDrop implements the pilorama.Forest interface.
 func (t *boltForest) TreeDrop(cid cidSDK.ID, treeID string) error {
 	return t.db.Batch(func(tx *bbolt.Tx) error {
+		if treeID == "" {
+			c := tx.Cursor()
+			prefix := bucketName(cid, treeID)
```
Comment (carpawell) [inline pkg/local_object_storage/pilorama/boltdb.go:632]: do we need a `bucketName` call here? is `[]byte(cid.EncodeToString())` enough and more verbose?

File: pkg/local_object_storage/pilorama/forest_test.go [hunk 4]
```diff
@@ -202,6 +204,19 @@ func testForestTreeDrop(t *testing.T, s Forest) {
 
 	_, err = s.TreeGetByPath(cid, trees[1], AttributeFilename, []string{"path"}, true)
 	require.NoError(t, err)
+
+	for i := range trees {
+		_, err := s.TreeAddByPath(d, trees[i], AttributeFilename, []string{"path"},
+			[]KeyValue{{Key: "TreeName", Value: []byte(trees[i])}})
+		require.NoError(t, err)
+	}
+	list, err := s.TreeList(cid)
```
Comment (carpawell) [inline pkg/local_object_storage/pilorama/forest_test.go:213]: check here that the `list` is not empty?

File: pkg/local_object_storage/pilorama/boltdb.go (hunk 3 above)
Comment (acid-ant) [reply pkg/local_object_storage/pilorama/boltdb.go]: Yes, looks better, added.

File: pkg/local_object_storage/pilorama/forest_test.go (hunk 4 above)
Comment (acid-ant) [reply pkg/local_object_storage/pilorama/forest_test.go:213]: Thanks, fixed.

**Review by carpawell — APPROVED** [review]

File: pkg/local_object_storage/pilorama/forest_test.go [hunk 5]
```diff
@@ -202,6 +204,20 @@ func testForestTreeDrop(t *testing.T, s Forest) {
 
 	_, err = s.TreeGetByPath(cid, trees[1], AttributeFilename, []string{"path"}, true)
 	require.NoError(t, err)
+
+	for i := range trees {
+		_, err := s.TreeAddByPath(d, trees[i], AttributeFilename, []string{"path"},
```
Comment (fyrchik) [inline pkg/local_object_storage/pilorama/forest_test.go:209]: Can we add some nodes for multiple container IDs and also check that all other trees are still present?

Comment (acid-ant) [reply pkg/local_object_storage/pilorama/forest_test.go]: Thanks, added.

**Review by fyrchik — APPROVED** [review]

**Pull request merged**